components/user-store/org.wso2.carbon.identity.user.store.configuration.deployer/src/main/java/org/wso2/carbon/identity/user/store/configuration/deployer/util/UserStoreUtil.java

```diff
@@ -118,6 +121,39 @@ public static Cipher getCipherOfSuperTenant() throws UserStoreException {
      */
     public static byte[] encrypt(byte[] plainTextBytes) throws CryptoException {
 
+        boolean isInternalKeyStoreEncryptionEnabled = false;
+        boolean isSymmetricKeyEncryptionEnabled = false;
+        ServerConfigurationService config =
+                UserStoreConfigComponent.getServerConfigurationService();
+        if (config != null) {
+            String encryptionKeyStore = config.getFirstProperty(UserStoreConfigurationConstants.ENCRYPTION_KEYSTORE);
+
+            if (INTERNAL_KEYSTORE.equalsIgnoreCase(encryptionKeyStore)) {
+                isInternalKeyStoreEncryptionEnabled = true;
+            }
+            String cryptoProvider = config.getFirstProperty(UserStoreConfigurationConstants.CRYPTO_PROVIDER);
+            if (UserStoreConfigurationConstants.SYMMETRIC_KEY_CRYPTO_PROVIDER.equalsIgnoreCase(cryptoProvider)) {
+                isSymmetricKeyEncryptionEnabled = true;
+            }
+        }
+
+        if (isInternalKeyStoreEncryptionEnabled && isSymmetricKeyEncryptionEnabled) {
+
+            throw new CryptoException(String.format("Userstore encryption can not be supported due to " +
+                            "conflicting configurations: '%s' and '%s'. When using internal keystore, assymetric crypto " +
+                            "provider should be used.", UserStoreConfigurationConstants.INTERNAL_KEYSTORE,
+                    UserStoreConfigurationConstants.SYMMETRIC_KEY_CRYPTO_PROVIDER));
+        } else if (isInternalKeyStoreEncryptionEnabled || isSymmetricKeyEncryptionEnabled) {
+
```

Comment:
lets remove this line as well

Reply:
Fixed with 57ad2c97fc9486e4803a66185de8851642574848